src/Tickets/Commerce/Admin_Tables/Orders_Table.php

```diff
@@ -676,26 +769,64 @@ protected function gateways_dropdown( $post_type ) {
 		$gateways_formatted = [
 			'' => esc_html__( 'All Gateways', 'event-tickets' ),
 		];
+
 		foreach ( $gateways as $gateway ) {
 			$gateways_formatted[ $gateway ] = ucfirst( $gateway );
 		}
 
-		$field = [
-			'type'    => 'dropdown',
-			'options' => $gateways_formatted,
+		?>
+		<label for="tec_tc_gateway-select" class="screen-reader-text"><?php esc_html_e( 'Filter By Gateway', 'event-tickets' ); ?></label>
+		<select name="tec_tc_gateway" id='tec_tc_gateway-select' class='tribe-dropdown' data-prevent-clear='true'>
+			<?php foreach ( $gateways_formatted as $key => $value ) : ?>
+				<option value="<?php echo esc_attr( $key ); ?>" <?php selected( $g, $key ); ?>><?php echo esc_html( $value ); ?></option>
+			<?php endforeach; ?>
+		</select>
+		<?php
+	}
+
+	/**
+	 * Displays a dropdown for filtering items in the list table by month.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $post_type The post type.
+	 *
+	 * @return void
+	 */
+	protected function post_parent_dropdown( $post_type ) {
+		/**
+		 * Filters whether to remove the 'Event' drop-down from the order list table.
+		 *
+		 * @since TBD
+		 *
+		 * @param bool   $disable   Whether to disable the drop-down. Default false.
+		 * @param string $post_type The post type.
+		 */
+		if ( apply_filters( 'tec_tc_orders_disable_post_parent_dropdown', false, $post_type ) ) {
+			return;
+		}
+
+		// Event options are being filtered in the Frontend after the user starts typing in the search box.
+		// Except for when the user has already filtered by an event. We take the event ID from the URL and add it to the dropdown.
+
+		$e = absint( tribe_get_request_var( 'tec_tc_events', 0 ) );
+
+		$event = $e ? get_post( $e ) : null;
+
+		$event = $event instanceof WP_Post ? $event : null;
+
+		$events_formatted = [
+			'' => esc_html__( 'All Events', 'event-tickets' ),
 		];
 
-		add_filter( 'tribe_field_start', '__return_empty_string' );
-		add_filter( 'tribe_field_end', '__return_empty_string' );
-		add_filter( 'tribe_field_div_start', '__return_empty_string' );
-		add_filter( 'tribe_field_div_end', '__return_empty_string' );
+		$events_formatted += $event ? [ (string) $event->ID => get_the_title( $event->ID ) ] : [];
```

Comment:
Let's skip the one line for a if($event) check to add the data to the array Otherwise, it will run the [loop](https://github.com/the-events-calendar/event-tickets/pull/3079/files#diff-cd5f0a73f32dd0d696f1cf3cb5b89c236eae6675e25a064f77150f1c6582895dR826) for the `[]` if that ever gets added,

Reply:
No it doesn't. Adding an empty array to an already existing array results to the original array.

```php
<?php

$test = [ '1' => 2 ];

$test += [];

var_dump( $test );
```

```php
array(1) {
  [1]=>
  int(2)
}
```